pyshelf/bulk_update/cleaner.py

```diff
@@ -0,0 +1,34 @@
+from pyshelf.bucket_update import utils
+
+
+class Cleaner(object):
+    """
+        Takes care of cleaning old documents from the Elasticsearch index.
+    """
+    def __init__(self, container):
+        self.container = container
+        self.config = self.container.config
+
+    def run(self):
+        """
+            Runs search index cleaning process.
+        """
+        id_list = []
+        bucket_container = None
+
+        for bucket in self.config["buckets"]:
+            # Creating bucket container per bucket
+            bucket.update({
+                "elasticsearch": self.config["elasticsearch"]
+            })
+            bucket_container = utils.create_bucket_container(bucket, self.container.logger)
+            path_list = bucket_container.search_updater.load_path_list()
+            self.container.logger.debug("Gathering existant artifacts for {0}".format(bucket["referenceName"]))
+
+            for path in path_list:
+                identity = bucket_container.resource_identity_factory.from_cloud_identifier(path)
+                id_list.append(identity.search)
+
+        update_manager = bucket_container.search_container.update_manager
+        response = update_manager.remove_unlisted_documents(id_list)
```

Comment:
How hard would it be to instead delete what is not in a particular artifactPath?  If it is not very simple I don't want to do it, but if so, I think it would be for efficient that having to query for all artifact paths in storage.  Do something like
```
# Extreme pseudo code
path_list.append(bucket["referenceName"] + "/artifact/*"

# later
elasticsearch.remove_paths_not_in_this_list(path_list)
```